cds/modules/home/decorators.py

```diff
@@ -24,14 +24,14 @@
 
 from functools import wraps
 
-from flask import session
+from flask import session, current_app
 from flask_login import current_user
 from invenio_cache import current_cache
 
 
 def has_flashes_or_authenticated_user():
     """Return True if there are pending flashes or user is authenticated."""
-    return '_flashes' in session or current_user.is_authenticated
+    return '_flashes' in session or current_user.is_authenticated or current_app.debug
```

Comment:
```python 
current_app.config.get('DEBUG')
``` 
☺️

Reply:
I checked the config in Flask and they say that some of the config variables are passed directly as attributes to the app in order to be read from there. So do you think is ok also like this? Here is the reference to the docs http://flask.pocoo.org/docs/0.12/config/#configuration-basics

Reply:
After checking the code, I think we can get rid of the function below (cached_unless_authenticated_or_flashes).
In `invenio-cache`, it is just a matter of setting the config var `CACHE_IS_AUTHENTICATED_CALLBACK` to the function above.
https://github.com/inveniosoftware/invenio-cache/blob/master/invenio_cache/decorators.py#L41

Reply:
Ok I am doing the refactor now.